.github/workflows/reusable-testing.yml

```
@@ -39,6 +39,17 @@ jobs:
         with:
           files: "composer.json, phpunit.xml.dist"
 
+      - name: Get composer.json PHP version
+        id: parse-composer
+        run: |
+          echo "COMPOSER_PHP_VERSION=$(jq '.require.php' composer.json | sed 's/[^0-9.|]//g' | jq -Rc 'split("|") | min | tonumber')" >> $GITHUB_OUTPUT;
+
+      - name: Exit early if matrix PHP version lower than composer.json required PHP version
+        run: if [[ $(bc -l <<< "$COMPOSER_PHP_VERSION > $MATRIX_PHP_VERSION") -eq 1 ]]; then exit 1; fi
```

```suggestion
        run: if [[ $(bc -l <<< "$COMPOSER_PHP_VERSION > $MATRIX_PHP_VERSION") -eq 1 ]]; then exit 0; fi
```

Is this a horrible idea?

---

That change would not exit the job (matrix instance run) – return 0 (exit success) is the same as any command being run in that step returning 0, and GitHub Actions continues to the next step on success. We have to return failure to exit the job.